Build task problems in the Problems tab have unresolved filepaths (VSC-276)

**Describe the bug**
After running the build task, the problems tab is populated with entries parsed from the output. 
However, clicking one of these results in a VS Code error: 

> Unable to open 'main.cpp': Unable to read file '/../main/main.cpp' (Error: Unable to resolve non-existing file '/../main/main.cpp').

Where the actual filepath is '/home/boarchuz/Projects/MyProject/main/main.cpp'

**To Reproduce**
In a project with any error, "Tasks - Run task" -> "Build - Build project". Click the resulting entry in the Problems tab.

**Solution**
The [default problemMatcher config](https://github.com/espressif/vscode-esp-idf-extension/blob/d1f5fd16b3e574a5e95726b93e7660aae915e760/templates/.vscode/tasks.json#L18) has "fileLocation": "absolute" while the output provides relative paths.

Example build output error:
> ../main/main.cpp: In function 'void app_main()':
> ../main/main.cpp:82:7: error: 'some_undeclared_variable' was not declared in this scope
>      ++some_undeclared_variable;
>        ^~~~~~~~~~~~~~~~~~~~~~~~

This can be solved by changing "fileLocation" to the following:
`"fileLocation": ["relative", "${workspaceFolder}/build"],`

The only thing I'm not 100% sure about is if that path is ideal. ie. Which path is the build output's "../main/main.cpp" actually relative to? Using "fileLocation": ["relative", "${workspaceFolder}"] would also work with an altered regexp to handle the "../".

There are many other tasks that seem to use the same problemMatcher config that may be affected too.

**Environment**
Linux Mint 19.3, VS Code 1.43.1, IDF Extension 0.1.4, IDF 'master' (v4.2-dev-701-g0ae960f2f)

--- a/templates/.vscode/tasks.json
+++ b/templates/.vscode/tasks.json
@@ -13,18 +13,32 @@
                     "PATH": "${config:idf.customExtraPaths}"
                 }
             },
-            "problemMatcher": {
-                "owner": "cpp",
-                "fileLocation": ["relative", "${workspaceFolder}/build"],
-                "pattern": {
-                    "regexp": "^(.*):(\\d+):(\\d+):\\s+(warning|error):\\s+(.*)$",
-                    "file": 1,
-                    "line": 2,
-                    "column": 3,
-                    "severity": 4,
-                    "message": 5
+            "problemMatcher": [
+                {
+                    "owner": "cpp",
+                    "fileLocation": ["relative", "${workspaceFolder}"],
+                    "pattern": {
+                        "regexp": "^\\.\\.(.*):(\\d+):(\\d+):\\s+(warning|error):\\s+(.*)$",
+                        "file": 1,
+                        "line": 2,
+                        "column": 3,
+                        "severity": 4,
+                        "message": 5
+                    }
+                },
+                {
+                    "owner": "cpp",
+                    "fileLocation": "absolute",
+                    "pattern": {
+                        "regexp": "^[^\\.](.*):(\\d+):(\\d+):\\s+(warning|error):\\s+(.*)$",
+                        "file": 1,
+                        "line": 2,
+                        "column": 3,
+                        "severity": 4,
+                        "message": 5
+                    }
                 }
-            },
+            ],
             "group": {
                 "kind": "build",
                 "isDefault": true
@@ -59,18 +73,32 @@
                     "PATH": "${config:idf.customExtraPaths}"
                 }
             },
-            "problemMatcher": {
-                "owner": "cpp",
-                "fileLocation": ["relative", "${workspaceFolder}/build"],
-                "pattern": {
-                    "regexp": "^(.*):(\\d+):(\\d+):\\s+(warning|error):\\s+(.*)$",
-                    "file": 1,
-                    "line": 2,
-                    "column": 3,
-                    "severity": 4,
-                    "message": 5
+            "problemMatcher": [
+                {
+                    "owner": "cpp",
+                    "fileLocation": ["relative", "${workspaceFolder}"],
+                    "pattern": {
+                        "regexp": "^\\.\\.(.*):(\\d+):(\\d+):\\s+(warning|error):\\s+(.*)$",
+                        "file": 1,
+                        "line": 2,
+                        "column": 3,
+                        "severity": 4,
+                        "message": 5
+                    }
+                },
+                {
+                    "owner": "cpp",
+                    "fileLocation": "absolute",
+                    "pattern": {
+                        "regexp": "^[^\\.](.*):(\\d+):(\\d+):\\s+(warning|error):\\s+(.*)$",
+                        "file": 1,
+                        "line": 2,
+                        "column": 3,
+                        "severity": 4,
+                        "message": 5
+                    }
                 }
-            },
+            ],
         },
         {
             "label": "Flash - Flash the device",
@@ -84,18 +112,32 @@
                     "PATH": "${config:idf.customExtraPaths}"
                 }
             },
-            "problemMatcher": {
-                "owner": "cpp",
-                "fileLocation": ["relative", "${workspaceFolder}/build"],
-                "pattern": {
-                    "regexp": "^(.*):(\\d+):(\\d+):\\s+(warning|error):\\s+(.*)$",
-                    "file": 1,
-                    "line": 2,
-                    "column": 3,
-                    "severity": 4,
-                    "message": 5
+            "problemMatcher": [
+                {
+                    "owner": "cpp",
+                    "fileLocation": ["relative", "${workspaceFolder}"],
+                    "pattern": {
+                        "regexp": "^\\.\\.(.*):(\\d+):(\\d+):\\s+(warning|error):\\s+(.*)$",
+                        "file": 1,
+                        "line": 2,
+                        "column": 3,
+                        "severity": 4,
+                        "message": 5
+                    }
+                },
+                {
+                    "owner": "cpp",
+                    "fileLocation": "absolute",
+                    "pattern": {
+                        "regexp": "^[^\\.](.*):(\\d+):(\\d+):\\s+(warning|error):\\s+(.*)$",
+                        "file": 1,
+                        "line": 2,
+                        "column": 3,
+                        "severity": 4,
+                        "message": 5
+                    }
                 }
-            },
+            ],
         },
         {
             "label": "Monitor: Start the monitor",

--- a/templates/.vscode/tasks.json
+++ b/templates/.vscode/tasks.json
@@ -15,7 +15,7 @@
             },
             "problemMatcher": {
                 "owner": "cpp",
-                "fileLocation": "absolute",
+                "fileLocation": ["relative", "${workspaceFolder}/build"],
                 "pattern": {
                     "regexp": "^(.*):(\\d+):(\\d+):\\s+(warning|error):\\s+(.*)$",
                     "file": 1,
@@ -61,7 +61,7 @@
             },
             "problemMatcher": {
                 "owner": "cpp",
-                "fileLocation": "absolute",
+                "fileLocation": ["relative", "${workspaceFolder}/build"],
                 "pattern": {
                     "regexp": "^(.*):(\\d+):(\\d+):\\s+(warning|error):\\s+(.*)$",
                     "file": 1,
@@ -86,7 +86,7 @@
             },
             "problemMatcher": {
                 "owner": "cpp",
-                "fileLocation": "absolute",
+                "fileLocation": ["relative", "${workspaceFolder}/build"],
                 "pattern": {
                     "regexp": "^(.*):(\\d+):(\\d+):\\s+(warning|error):\\s+(.*)$",
                     "file": 1,

